.github/workflows/CI_release.yml

```
@@ -0,0 +1,85 @@
+name: CI on release
```

I think this is the [Continuous delivery](https://www.redhat.com/en/topics/devops/what-is-ci-cd) part of our release workflow.
```suggestion
name: CD on release tag
```
The name of the YAML file should also be changed to "CD_release.yml", I think

---

I have changed the file name (and also changed it on the trusted publisher list on testpypi)